app/models/miq_provision_virt_workflow.rb

```diff
@@ -216,15 +216,19 @@ def available_vlans_and_hosts(options = {})
       rails_logger('allowed_vlans', 0)
       # TODO: Use Active Record to preload this data?
       MiqPreloader.preload(hosts, :switches => :lans)
-      hosts.each do |h|
-        h.lans.each { |l| vlans[l.name] = l.name unless l.switch.shared? }
-      end
+      load_allowed_vlans(hosts, vlans)
       rails_logger('allowed_vlans', 1)
     end
 
     return vlans, hosts
   end
 
+  def load_allowed_vlans(hosts, vlans)
+    hosts.each do |h|
+      h.lans.each { |l| vlans[l.name] = l.name if @vlan_options[:dvs] || !l.switch.shared? }
```

Comment:
It shouldn't break VMware because we override the whole `available_vlans_and_hosts` method [here](https://github.com/agrare/manageiq-providers-vmware/blob/335bab6603e8cb64e7090f1785ba380cb6cf0c10/app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb#L128-L145)

Why did you add back in the `if @vlan_options[:dvs]`?  I removed this in this PR https://github.com/ManageIQ/manageiq/pull/14292 because we moved all dvs handling to vmware here https://github.com/ManageIQ/manageiq-providers-vmware/pull/25